packages/admin-next/dashboard/src/v2-routes/promotions/promotion-list/loader.ts

```diff
@@ -0,0 +1,25 @@
+import { AdminPromotionsListRes } from "@medusajs/medusa"
+import { Response } from "@medusajs/medusa-js"
+import { QueryClient } from "@tanstack/react-query"
+
+import { adminPromotionKeys } from "../../../lib/api-v2"
+import { medusa, queryClient } from "../../../lib/medusa"
+
+const promotionsListQuery = () => ({
+  queryKey: adminPromotionKeys.list({ limit: 20, offset: 0 }),
+  // Question: whats the v2 equivalenet of this?
+  queryFn: async () => medusa.admin.discounts.list({ limit: 20, offset: 0 }),
+})
+
+// Question: Are we using this anywhere magically? Can this file be removed?
```

Comment:
q: is this file needed? What exactly does it do?

Reply:
cc. @kasperkristensen @fPolic i have added some comments in the PR if you guys can help here? Thanks ❤️

Reply:
This is a react router pattern for preloading data for the route. Each route can define a main component and a loader, data loaded here will be available in the `Component` exported from the index.

https://reactrouter.com/en/main/route/loader